src/oracle/connection/stmt.rs

```diff
@@ -122,7 +157,20 @@ impl Statement {
         }
     }
 
-    pub fn run(&self) -> QueryResult<()> {
+    pub fn check_error_sql(
+        error_handle: *mut ffi::OCIError,
+        status: i32,
+        sql: &String,
+        action: String,
```

Comment:
Do we really need a `String` here, or could we just use a `&str` to skip some allocations?

(This would allow removing all those `.to_string()` calls while calling this function)